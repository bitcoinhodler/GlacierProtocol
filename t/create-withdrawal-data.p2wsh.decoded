{
  "txid": "5d56f92d823a35e5cebc23e5d918377eab1120d468c09d4c9bc9294676a3a482",
  "hash": "1fdb48a6daeac066891fedcbcbc27996d3b5905bb35260dac2334820f9b5043e",
  "version": 2,
  "size": 370,
  "vsize": 154,
  "weight": 616,
  "locktime": 0,
  "vin": [
    {
      "txid": "61c0a01d78717b6e17f2dc4068090f044365f7bc22a2cce82038d76bff6d8171",
      "vout": 0,
      "scriptSig": {
        "asm": "",
        "hex": ""
      },
      "txinwitness": [
        "",
        "30440220358fff108e30dc3071e835d986f359bedebec3485d36b083add0cc4bf29648f60220090f63e00cc326a0400e4f9540edbae1f79dd9bcbb6e1cc5c949efa42cd4cf6301",
        "304402207ac62e00bcab502f9c93a7f8b9f4d853ffcc26a421daa1871108dacc9b0507cc022037b61e4ca61a200f7244fe69640d787b3b85d3cb290fc8d26c247a1ec961d34a01",
        "522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.04999692,
      "n": 0,
      "scriptPubKey": {
        "asm": "0 a410c764ac8da12b578a06027f11e02f07b7e1d3",
        "desc": "addr(bcrt1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnrkhp23)#5da9chcd",
        "hex": "0014a410c764ac8da12b578a06027f11e02f07b7e1d3",
        "address": "bcrt1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnrkhp23",
        "type": "witness_v0_keyhash"
      }
    }
  ]
}
