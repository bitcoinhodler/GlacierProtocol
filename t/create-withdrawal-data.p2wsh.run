#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
bcrt1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q2wgvlj
522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae
bcrt1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnrkhp23
1
020000000001018344c09044c78e96007f879f161a3a2d624afba7263f347269a7cea6feba62ff010000001716001410ed28fa7173084dfd6cba88383e031e8e7a5e9dffffffff02404b4c000000000022002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8dba19e0000000000160014dacd1afaae47b1c4b9adbbb261df55ebf898b3800247304402202c65de8392a9fc66713f5e154e0022dca84ff89acd152f008ea0a24b73b7c38902204a2ff68838cf880568cb6100ccff23e52e6f9940d3d77557b4795e9f5001fbc5012102c745d1c5e15240939f10b29836741c9b809f6874302f633c21bd95671e7d7b9700000000
2
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
2
y

y
INPUT
