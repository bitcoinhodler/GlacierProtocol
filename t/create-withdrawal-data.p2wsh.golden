Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.05000000 BTC

How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00000394 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00000394, you have 0.04999606 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to bcrt1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnrkhp23 (leave blank to withdraw all funds stored in these unspent transactions): With no change output, the transaction fee is reduced, and 0.04999692 BTC will be sent to your destination.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.05000000 BTC in unspent inputs from cold storage address bcrt1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q2wgvlj
0.04999692 BTC going to destination address bcrt1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnrkhp23
Fee amount: 0.00000308

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4


Confirm? (y/n): 
Calculating transaction...


Sufficient private keys to execute transaction?
True

Raw signed transaction (hex):
0200000000010171eba4ea0f932ba9d1734b4e67b4dbdb40a7dd9b07c7f9df884ee68fc1e5bbad0000000000ffffffff010c4a4c0000000000160014a410c764ac8da12b578a06027f11e02f07b7e1d304004730440220218480f6701fcc31066da334b22efa1c64d2a64a53c1b7751559c2062f9a150a02203b987ccf310a0d097a54207d6f502f4dba4e5ac35059ec569d2ce4269486ee730147304402201931993fb10639aa822a7243bb94b59917563d7c180494310a737b41fe1b582302205efefd5305ce4bd4d31f0dc8c1814759110c329affcf8af99b46781ff6db1142018b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae00000000

Transaction fingerprint (md5):
2264b7b792021c36919e788e19927a7b
QR code for transaction written to transaction.png
